include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+
+namespace spirv
+{
+template<typename M, typename T, typename StorageClass>
+[[vk::ext_instruction(/*spv::OpAccessChain*/65)]]
+vk::SpirvOpaqueType </* OpTypePointer*/ 32,StorageClass,M> accessChain(
+    [[vk::ext_reference]] vk::SpirvOpaqueType </* OpTypePointer*/ 32,
+    StorageClass,T>base,
+    [[vk::ext_literal]] uint32_t index0
+);
+
+template<class T, class U>
+[[vk::ext_instruction( /*spv::OpBitcast*/124)]]
+T bitcast(U);
+
+template<typename T> // integers operate on 2s complement so same op for signed and unsigned
+[[vk::ext_instruction(/*spv::OpAtomicIAdd*/234)]]
+T atomicIAdd([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+}
```

Comment:
we already have the `spirv_intrinsics` folder that has these (except `accessChain` thats new)

Reply:
Should the accessChain go in the spirv_intrinsics folder then?

Reply:
yes, I'm not sure how the syntax should look like yet... it probably shouldn't have the vk::ext_reference